estimagic/visualization/comparison_plot.py

```diff
@@ -0,0 +1,440 @@
+        lower=lower,
+        upper=upper,
+        rect_widths=rect_widths,
+        width=width,
+    )
+
+    plots_with_callbacks = _add_callbacks(
+        source_dict=source_dict,
+        figure_dict=figure_dict,
+        glyph_dict=glyph_dict,
+        model_classes=sorted(df["model_class"].unique()),
+    )
+
+    grid = gridplot(plots_with_callbacks, toolbar_location="right", ncols=1)
+    show(grid)
+    return df, grid, plots_with_callbacks
+
+
+# ===========================================================================
+# DATA PREP FUNCTIONS
+# ===========================================================================
+
+
+def _df_with_all_results(res_dict):
+    """
+    Build the DataFrame combining all DataFrames from the results dictionary.
+
+    Args:
+        res_dict (dict): see comparsion_plot docstring
+    """
+    df = pd.DataFrame()
+    for model, model_dict in res_dict.items():
+        result_df = _prep_result_df(model_dict, model)
+        df = df.append(result_df, sort=False)
+
+    if "group" not in df.columns:
+        df["group"] = "all"
+    if "conf_int_upper" not in df.columns:
+        df["conf_int_upper"] = np.nan
+    if "conf_int_lower" not in df.columns:
+        df["conf_int_lower"] = np.nan
+
+    return df.reset_index(drop=True)
+
+
+def _prep_result_df(model_dict, model):
+    result_df = model_dict["result_df"].reset_index()
+    result_df["model"] = model
+    name_cols = [x for x in ["group", "name"] if x in result_df.columns]
+    result_df["full_name"] = result_df[name_cols].apply(
+        lambda x: index_element_to_string(tuple(x)), axis=1
+    )
+    if "model_class" in model_dict.keys():
+        model_class = model_dict["model_class"]
+        result_df["model_class"] = model_class
+    else:
+        result_df["model_class"] = "no class"
+    return result_df
+
+
+def _create_bounds_and_rect_widths(df):
+    """Calculate lower and upper bounds and the width of the "histogram" bars."""
+    grouped = df.groupby("group")
+    upper = grouped[["conf_int_upper", "final_value"]].max().max(axis=1)
+    lower = grouped[["conf_int_lower", "final_value"]].min().min(axis=1)
+    rect_widths = 0.02 * (upper - lower)
+
+    return lower, upper, rect_widths
+
+
+def _determine_plot_heights(df, figure_height):
+    grouped = df.groupby("group")
+    figure_height = _determine_figure_height(df, figure_height)
+    nr_params = grouped["full_name"].unique().apply(len)
+    height_shares = nr_params / nr_params.sum()
+    return (height_shares * figure_height).astype(int)
+
+
+def _determine_figure_height(df, figure_height):
+    if figure_height is None:
+        nr_models = len(df["model"].unique())
+        nr_params = len(df["full_name"].unique())
+        figure_height = 8 * max(min(nr_models, 60), 10) * nr_params
+    return figure_height
+
+
+def _add_plot_specs_to_df(df, rect_widths, lower, upper, color_dict):
+    """Add color column, dodge column and binned_x colomun to *df*."""
```

Comment:
typo

Reply:
Fixed in 604f9b37a2dc4bc5123ddc29a8fceba432d60334.